### PR TITLE
File restructure by providers for add_cloud_metadata

### PR DESCRIPTION
**Pull request by athom**
This PR is a follow up issue for #4023 , focus on adjusting the files structure according the cloud providers. makes it easier to spot which parts are specific to a provider and which is the shared code.

### CODE REVIEW

File: libbeat/processors/add_cloud_metadata/provider_aws_ec2.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package actions
```
Comment (ruflin) [inline libbeat/processors/add_cloud_metadata/provider_aws_ec2.go]: Not related to this change, but why is this package called `actions`? @andrewkroh

Comment (andrewkroh) [inline libbeat/processors/add_cloud_metadata/provider_aws_ec2.go]: At one point there was an actions folder and I think we decided in the original PR for this processor to move it into its own folder. Looks like I did the file move, but never updated the actual package name.

Comment (athom) [inline libbeat/processors/add_cloud_metadata/provider_aws_ec2.go]: Should I make some related changes by the way?

Comment (andrewkroh) [inline libbeat/processors/add_cloud_metadata/provider_aws_ec2.go]: That would be great. Thanks

Comment (athom) [inline libbeat/processors/add_cloud_metadata/provider_aws_ec2.go]: Change is made, also applied on `add_locale` package too.
BTW, could the reviewdog be taught to recognize such convention?

Comment (ruflin) [inline libbeat/processors/add_cloud_metadata/provider_aws_ec2.go]: Thanks.

For reviewdog, @7AC perhaps knows more.

**Comment by ruflin** [comment]
jenkins, test  it

File: libbeat/processors/add_cloud_metadata/provider_aws_ec2.go (hunk 1 above)
Comment (7AC) [inline libbeat/processors/add_cloud_metadata/provider_aws_ec2.go]: Reviewdog just does what `golint` does

**Pull request merged**